File: abc/abc350/a/main.php
<?php

$input = trim(fgets(STDIN));

$num = substr($input, 3, 3);

// print_r($num);

if($num == 316 || $num >= 350 || $num == 000){
    echo "No\n";
    exit;
}else{
    echo "Yes\n";
    exit;
}

File: abc/abc350/a/main.py
S = input()
A=[]

for i in range(9):    
    w=str(i+1)
    A.append("ABC"+"0"+"0"+w)

for i in range(9,99):    
    w=str(i+1)
    A.append("ABC"+"0"+w)

for i in range(99,349):
    w=str(i+1)
    A.append("ABC"+w)

A.remove("ABC316")

if S in A:
  print("Yes")
else:
  print("No")